browser/ui/webui/brave_rewards_ui.cc

```diff
@@ -1079,6 +1098,72 @@ void RewardsDOMHandler::OnContributionSaved(
       "brave_rewards.onContributionSaved", result);
 }
 
+void RewardsDOMHandler::GetPendingContributions(
+    const base::ListValue* args) {
+  if (rewards_service_) {
+    rewards_service_->GetPendingContributionsUI(base::Bind(
+          &RewardsDOMHandler::OnGetPendingContributions,
+          weak_factory_.GetWeakPtr()));
+  }
+}
+
+void RewardsDOMHandler::OnGetPendingContributions(
+    std::unique_ptr<brave_rewards::PendingContributionInfoList> list) {
+  if (web_ui()->CanCallJavascript()) {
+    auto contributions = std::make_unique<base::ListValue>();
+    for (auto const& item : *list) {
+      auto contribution = std::make_unique<base::DictionaryValue>();
+      contribution->SetString("publisherKey", item.publisher_key);
+      contribution->SetBoolean("verified", item.verified);
+      contribution->SetString("name", item.name);
+      contribution->SetString("provider", item.provider);
+      contribution->SetString("url", item.url);
+      contribution->SetString("favIcon", item.favicon_url);
+      contribution->SetDouble("amount", item.amount);
+      contribution->SetInteger("addedDate", item.added_date);
+      contribution->SetInteger("category", item.category);
+      contribution->SetString("viewingId", item.viewing_id);
+      contribution->SetInteger("expirationDate", item.expiration_date);
+      contributions->Append(std::move(contribution));
+    }
+
+    web_ui()->CallJavascriptFunctionUnsafe("brave_rewards.pendingContributions",
+                                           *contributions);
+  }
+}
+
+void RewardsDOMHandler::RemovePendingContribution(
+    const base::ListValue* args) {
+  if (rewards_service_) {
+    std::string publisher_key;
+    std::string viewing_id;
+    int added_date;
+    args->GetString(0, &publisher_key);
+    args->GetString(1, &viewing_id);
+    args->GetInteger(2, &added_date);
+    rewards_service_->RemovePendingContributionUI(
+        publisher_key,
+        viewing_id,
+        static_cast<uint64_t>(added_date));
+  }
+}
+
+void RewardsDOMHandler::RemoveAllPendingContributions(
+    const base::ListValue* args) {
+  if (rewards_service_) {
+    rewards_service_->RemoveAllPendingContributionsUI();
+  }
+}
+
+void RewardsDOMHandler::OnPendingContributionRemoved(
+    brave_rewards::RewardsService* rewards_service,
+    int32_t result) {
```

Comment:
We should be having `#include <stdint.h>` for (u)intx_t types

Reply:
done